### DISCUSSION TITLE
Cómo tratar datos no válidos?

### DISCUSSION THREAD

**Discussion by ypne** [IIC2413/Syllabus-2024-2#48] [General]
Se entiende como dato no válido a un dato que no cumple con las restricciones indicadas en el enunciado, existen varios casos
1.- se pide un int y viene un string
2.- se pide un valor específico y viene algo fuera de rango.
El objetivo es perder el minimo de datos posibles por lo que hay varias estrategias:
a.- si viene un. dato NULO en un atributo no nulo, se puede rellenar con un valor que indique esta condición por ejemplo el número de alumno se puede rellenar con 0 o un string con el texto "NULO"
b.- si en un campo enumerado como calificación viene nulo se puede marcar con "X" que es un valor fuera del rango de la enumeración
c.- si en el telefono vienen caracteres no numéricos se puede rellenar con "0000000" no 0
d.- si hay un campo cambiado de atributo, por ejemplo DIURNO en el campo jerarquía, se puede corregir y ponerlo en el atributo correcto
Ustedes pueden usar otras estrategias siempre que eviten borrar datos (excepto que no haya alternativa) y documentarlo en el informe.

**Comment by sbarriag** [comment]
Hola! en base a lo mismo queria preguntar si al verificar el número de teléfono, tengo que ver si es un número de "largo 9" ?? y de ser así y por ejemplo si alguien tiene número 123 lo reemplazo de la misma manera que si hubiera un error de caracteres como por ejemplo "0000000" o hay que hacer distinción de esos errores también?? 

Muchas gracias

**Reply by TomasKetterer** [reply]
Hola! Puedes hacerlo siempre que lo justifiques en tu informe, en este caso se puede ver que los números tienen formato 9XXXXXXXX, por lo que si algún número no cumple con eso, puedes justificarlo y manipular esos datos :)

**Reply by ypne** [reply]
Hola
Aclaración
Los teléfonos en chile pueden comenzar con los números 2 al 9, pero en esta etapa solo se está pidiendo que se valide el largo de 9 dígitos.

**Comment by cataruze** [comment]
Hola, no me quedó muy claro que hacer si tengo un valor inválido, por ejemplo nota 8, tendría que tratarlo igual a que si fuera un nulo? Es decir poder un valor no válido para llenar la casilla (“X”)?

**Reply by ypne** [reply]
ver discussion #86 [https://github.com/IIC2413/Syllabus-2024-2/discussions/86](url)

**Comment by ypne** [comment]
Hola
 Si el dato se puede corregir, se corrige
 Si no se puede corregir se puede marcar como malo (por ejemplo 000000000 para los teléfonos) o se elimina
 La opción que tomen la deben documentar en el informe.
